src/peer_manager.rs

```diff
@@ -325,42 +378,188 @@ impl PeerManager {
     }
 
     /// Wraps the routing table function of the same name and maps `XorName`s to `PublicId`s.
-    pub fn expect_add_to_our_section(&self,
-                                     expected_name: &XorName,
-                                     our_public_id: &PublicId)
-                                     -> Result<Vec<Section>, RoutingTableError> {
-        self.routing_table.validate_joining_node(expected_name)?;
-        let sections = self.routing_table.all_sections();
-        let mut result = vec![];
-        for (prefix, names) in sections {
-            let mut public_ids = vec![];
-            for name in names {
-                if name == *our_public_id.name() {
-                    public_ids.push(*our_public_id);
-                } else if let Some(peer) = self.peer_map.get_by_name(&name) {
-                    public_ids.push(*peer.pub_id())
+    ///
+    /// Returns:
+    ///
+    /// * Err()             If already handling a joining request
+    /// * Ok(our_section)   If needs to carry out resource proof evaluate
+    pub fn expect_join_our_section(&mut self,
+                                   expected_name: &XorName,
+                                   client_auth: &Authority<XorName>,
+                                   our_public_id: &PublicId)
+                                   -> Result<Vec<PublicId>, RoutingError> {
+        if let Some(Candidate { ref name, ref insertion_time, .. }) = self.candidate {
+            if name == expected_name ||
+               insertion_time.elapsed() < Duration::from_secs(RESOURCE_PROOF_APPROVE_TIMEOUT_SECS) {
+                return Err(RoutingError::AlreadyHandlingJoinRequest);
+            }
+        }
+
+        let names = self.routing_table.expect_join_our_section(expected_name)?;
+
+#[cfg(not(feature = "use-mock-crust"))]
+        let seed = vec![rand::random::<u8>(); 10];
```

Comment:
This just repeats the same random `u8` 10 times though.  The way that's used by the `resource_proof` crate, it'd be equivalent to `vec![rand::random::<u8>(); 1]`.

Shouldn't this be more like `let seed = thread_rng().gen_iter().take(10).collect();`?